dapp/src/utils/orangekit/ledger-live/bitcoin-provider.ts

```diff
@@ -0,0 +1,323 @@
+import {
+  Account,
+  defaultLogger,
+  WalletAPIClient,
+  WindowMessageTransport,
+} from "@ledgerhq/wallet-api-client"
+import { UserRejectedRequestError } from "viem"
+import {
+  Balance,
+  BitcoinAddress,
+  BitcoinTxHash,
+  OrangeKitBitcoinWalletProvider,
+} from "@orangekit/react/src/wallet/bitcoin-wallet-provider"
+import {
+  BitcoinNetwork,
+  AcreBitcoinProvider,
+  BitcoinSignatureHelper,
+  SafeTransactionData,
+} from "@acre-btc/sdk"
+import {
+  AcreMessage,
+  AcreMessageType,
+  AcreModule,
+} from "@ledgerhq/wallet-api-acre-module"
+import BigNumber from "bignumber.js"
+
+export type TryRequestFn<T> = (fn: () => Promise<T>) => Promise<T>
+
+function tryRequest<T>(): TryRequestFn<T> {
+  return async (fn) => {
+    try {
+      return await fn()
+    } catch (error: unknown) {
+      const err = error as Error
+
+      // We want to throw the same error for all connectors when user rejects
+      // the request.
+      if (
+        err?.message === "UserRefusedOnDevice" ||
+        err?.message === "Canceled by user"
+      ) {
+        throw new UserRejectedRequestError(err)
+      }
```

Comment:
If we return the same error for all connectors, do we still need to use the `ledgerLive.didUserRejectRequest` call ?

https://github.com/thesis/acre/blob/0d137f162470c1dc67e5b8a7005a3168d223ca2b/dapp/src/components/TransactionModal/ActiveStakingStep/DepositBTCModal.tsx#L54-L66

Reply:
This error occurs when the user closes the transaction/sign message window see the video below. So I wonder if we can't add this error message to the `tryRequest` function to remove the `ledgerLive.didUserRejectRequest` function. Of course, this is not a blocker but some later improvement.

https://github.com/user-attachments/assets/4ade40a8-618b-4198-9d3c-d065af1a4869